src/v2/guide/typescript.md

```diff
@@ -187,3 +187,41 @@ const Component = Vue.extend({
 ```
 
 If you find type inference or member completion isn't working, annotating certain methods may help address these problems. Using the `--noImplicitAny` option will help find many of these unannotated methods.
+
+
+
+## Annotating Props
+
+```ts
+import Vue, { PropType } from 'vue'
+
+interface ComplexMessage { 
+  title: string,
+  okMessage: string,
+  cancelMessage: string
+}
+const Component = Vue.extend({
+  props: {
+    name: String,
+    success: { type: String },
+    callback: { 
+      type: Function as PropType<() => void>
+    },
+    message: {
+      type: Object as PropType<ComplexMessage>,
+      required: true,
+      validator (message: ComplexMessage) {
+        return !!message.title;
+      }
+    },
+    notificationMessage: {
+      type: Object,
+      validation: (message) => {
+        return !!message.title;
+      } 
+    } as PropValidator<ComplexMessage>
```

Comment:
Just wanted to provide other ways to allow forcing the PropValidator, when you don't send the validation parameter type typescript has some problems to get type explicity, it will complain about implicit 'any'.
Other way to fix this would be annotate the validation with `as (message: ComplexMessage)=>boolean` or simply set the type on the parameter.

Reply:
Then let's just annotate validator parameter 🙂

Reply:
there is no need for that example then 🙂